File: Coding/Java/Set3/PerfectNumber.java
package Set3;



public class PerfectNumber {

    public static boolean isPerfectNumber(int num){
        int sum_divisors  = 0;
        int initial_divisor = 2;
        while(initial_divisor <= Math.sqrt(num)){
            if(num % initial_divisor == 0) {
                sum_divisors += initial_divisor;
                sum_divisors += num/initial_divisor;
            }
            initial_divisor++;
        }
        return num == sum_divisors+1;
    }

    public static void main(String[] args) {
        System.out.println(isPerfectNumber(15));
    }
}

File: Coding/Python/Set3/PerfectNumber.py
import math


def perfect_number(num):
    sum_divisors = 1
    for val in range(2, int(math.sqrt(num)) + 1):
        if num % val == 0:
            sum_divisors += val
            sum_divisors += num//val
    return num == sum_divisors


print(perfect_number(8))
